src/lib/mappers/charge-purpose.js

```diff
@@ -0,0 +1,111 @@
+'use strict';
+
+const { createMapper } = require('../object-mapper');
+
+const ChargePurpose = require('../models/charge-purpose');
+const DateRange = require('../models/date-range');
+const camelCaseKeys = require('../camel-case-keys');
+const purposePrimaryMapper = require('./purpose-primary');
+const purposeSecondaryMapper = require('./purpose-secondary');
+const purposeUseMapper = require('./purpose-use');
+const abstractionPeriodMapper = require('./abstraction-period');
+const dateRangeMapper = require('./date-range');
+const helpers = require('./lib/helpers');
+
+const timeLimitedDateMapper = (startDate, endDate) =>
+  new DateRange(startDate, endDate);
+
+const dbToModelMapper = createMapper()
+  .map('chargePurposeId').to('id')
+  .copy(
+    'source',
+    'season',
+    'loss',
+    'authorisedAnnualQuantity',
+    'billableAnnualQuantity',
+    'description',
+    'chargeElementId',
+    'isSection127AgreementEnabled'
+  )
+  .map().to('abstractionPeriod', abstractionPeriodMapper.dbToModel)
```

Comment:
As above, mapping to undefined values feels weird/wrong. Ignore me if you're sure this is right.

Reply:
yeah same reason as above